staging/src/k8s.io/kube-apiextensions-server/pkg/controller/status/naming_controller.go

```diff
@@ -236,6 +236,31 @@ func (c *NamingConditionController) sync(key string) error {
 		return err
 	}
 
+	crd := &apiextensions.CustomResourceDefinition{}
+	if err := apiextensions.DeepCopy_apiextensions_CustomResourceDefinition(inCustomResourceDefinition, crd, cloner); err != nil {
+		return err
+	}
+
+	for _, condition := range crd.Status.Conditions {
```

Comment:
Oh yes, that should be enough. Key is the name. I can extract the group name from the key and requeue the CRDs in that group.